python/ray/train/v2/_internal/execution/context.py

```diff
@@ -24,20 +24,30 @@
 logger = logging.getLogger(__file__)
 
 
-@dataclass
+@dataclass(frozen=True)
 class TrainRunContext:
     """Holds the metadata and context for the current training run."""
 
-    # TODO: Make this dataclass immutable after refactoring the train context.
-
     # The unique ID of the training run.
     run_id: str = field(init=False, default_factory=lambda: uuid.uuid4().hex)
 
     # The run configuration for the current training run.
     run_config: RunConfig
 
-    # TODO: Add more fields that are shared across all workers and controllers.
-    # For example, StorageContext, ScalingConfig, etc.
+    # The configuration passed to the training function.
+    train_loop_config: Optional[Dict[str, Any]]
```

Comment:
should we also add some metadata like ray version/commit, backend version (torch version)?